### PR TITLE
Support reverse lookups from the host file.

### PR DESCRIPTION
**Pull request by cmcfarlen**
Refactor host file code to make it more testable and unify lookups
Add unit test for HostFile

### CODE REVIEW

**Comment by cmcfarlen** [comment]
Fixes #9025

**Comment by cmcfarlen** [comment]
While this supports reverse lookups from the hosts file, there is some issue with deterministic selection of the name that is resolved for a given IP since the order depends on the appearance of the file as well as how we parse the hosts file.  Currently the unit test is broken because of this nondeterministic behavior.  Should we try to guarantee a deterministic resolution or should we just return any host with the assigned IP?  I'm not sure how ATS handles multiple hosts returned from a reverse lookup currently.

Should this "reverse lookup from the hosts file" be a configurable feature?

**Comment by SolidWallOfCode** [comment]
Is it indeterminate because an unordered map is used? I think it would be fine to say which ever name is first associated with an IP address is the reverse lookup name.

File: iocore/hostdb/HostFile.cc [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  limitations under the License.
+ */
+
+#include "HostFile.h"
+#include "P_HostDBProcessor.h"
+
+/* Container for temporarily holding data from the host file. For each FQDN there is a vector of IPv4
+ * and IPv6 addresses. These are used to generate the HostDBRecord instances that are stored persistently.
+ */
+using HostAddrMap = std::unordered_map<std::string_view, std::tuple<std::vector<IpAddr>, std::vector<IpAddr>>>;
```
Comment (SolidWallOfCode) [inline iocore/hostdb/HostFile.cc:30]: If you're going to split it anyway, you might as well use `Ip4Addr` and `Ip6Addr`.

**Comment by SolidWallOfCode** [comment]
I wonder if `Lexicon` would work in this situation. It supports primary and secondary names vs. a numeric value, which seems like the situation here.

**Comment by bryancall** [comment]
[approve ci autest]

**Review by SolidWallOfCode — APPROVED** [review]

**Review by maskit — CHANGES_REQUESTED** [review]
Is this for 10-Dev or not? It's marked as 10-Dev but the target branch is master.

**Comment by cmcfarlen** [comment]
> Is this for 10-Dev or not? It's marked as 10-Dev but the target branch is master.

I moved it to 10.0 milestone but idk how to remove it from the 10-dev project.

**Comment by bneradt** [comment]
> > Is this for 10-Dev or not? It's marked as 10-Dev but the target branch is master.
> 
> I moved it to 10.0 milestone but idk how to remove it from the 10-dev project.

I removed it. I just uncheck-marked the 10-Dev project.

**Comment by cmcfarlen** [comment]
> > > Is this for 10-Dev or not? It's marked as 10-Dev but the target branch is master.
> > 
> > 
> > I moved it to 10.0 milestone but idk how to remove it from the 10-dev project.
> 
> I removed it. I just uncheck-marked the 10-Dev project.

Thanks, I must not have that permission as I didn't see a checkmark to uncheck!

**Pull request merged**